policies/gac/networks.py

```diff
@@ -244,41 +253,58 @@ def _supervised_forward(self, state, taus, actions):
 
 
 class StochasticActor(IQNSuperClass):
-    def __init__(self, action_dim, n_basis_functions=64):
+    def __init__(self, state_dim, action_dim, n_basis_functions=64):
         """
         The IQN stochasitc action generator, takes state and tau (random vector) as input, and output
         the next action. This generator is not in an autoregressive way, i.e. the next action is
         generated as a whole, instead of one dimension by one dimension.
 
         Class Args:
+            state_dim (int): the dimensionality of the state vector
             action_dim (int): the dimensionality of the action vector
             n_basis_functions (int): the number of basis functions for noise embedding.
         """
         super(StochasticActor, self).__init__()
-        self.l1 = tf.keras.layers.Dense(420)
-        cosine_hidden_size = 420//action_dim
-        assert cosine_hidden_size * action_dim == 420
-        self.phi = CosineBasisLinear(n_basis_functions, cosine_hidden_size)
-        self.l2 = tf.keras.layers.Dense(200)
-        self.l3 = tf.keras.layers.Dense(action_dim)
+        self.module_type = 'StochasticActor'
+        self.noise_embed_dim = 400 // action_dim
+        self.action_dim = action_dim
+
+        self.state_embedding_layer = tf.keras.layers.Dense(
+            self.noise_embed_dim * self.action_dim,
+            activation=tf.keras.layers.LeakyReLU(alpha=0.01),
```

Comment:
why set alpha=0.01 instead of leave it as default?

Reply:
The default alpha in tf is 0.3, while in pytorch is 0.01. This is actually pretty big difference...My experience is that 0.01 works better, which is also consistent with their pytorch code.

Reply:
sounds good.